sdk/keyvault/keyvault/src/keys/index.ts

```diff
@@ -0,0 +1,476 @@
+   * @param [options] The optional parameters
+   * @returns Promise<DeletedKey>
+   */
+  public async deleteKey(
+    keyName: string,
+    options?: RequestOptionsBase
+  ): Promise<DeletedKey> {
+    const response = await this.client.deleteKey(this.vaultBaseUrl, keyName, options);
+    return this.getKeyFromKeyBundle(response);
+  }
+
+  /**
+   * The UPDATE operation changes specified attributes of an existing stored key. Attributes that
+   * are not specified in the request are left unchanged. The value of a key itself cannot be
+   * changed. This operation requires the keys/set permission.
+   * @summary Updates the attributes associated with a specified key in a given key vault.
+   * @param keyName The name of the key.
+   * @param keyVersion The version of the key.
+   * @param [options] The optional parameters
+   * @returns Promise<Key>
+   */
+  public async updateKeyAttributes(
+    keyName: string,
+    keyVersion: string,
+    options?: UpdateKeyOptions
+  ): Promise<Key> {
+    if (options) {
+      let unflattenedAttributes = { enabled: options.enabled, notBefore: options.notBefore, expires: options.expires };
+      let unflattenedOptions = { ...options, ...(options.requestOptions ? options.requestOptions : {}), keyAttributes: unflattenedAttributes };
+      delete unflattenedOptions.enabled;
+      delete unflattenedOptions.notBefore;
+      delete unflattenedOptions.expires;
+      delete unflattenedOptions.requestOptions;
+
+      const response = await this.client.updateKey(
+        this.vaultBaseUrl,
+        keyName,
+        keyVersion,
+        unflattenedOptions
+      );
+      return this.getKeyFromKeyBundle(response);
+    } else {
+      const response = await this.client.updateKey(
+        this.vaultBaseUrl,
+        keyName,
+        keyVersion,
+        options
+      );
+      return this.getKeyFromKeyBundle(response);
+    }
+  }
+
+  /**
+   * The GET operation is applicable to any key stored in Azure Key Vault. This operation requires
+   * the keys/get permission.
+   * @summary Get a specified key from a given key vault.
+   * @param keyName The name of the key.
+   * @param [options] The optional parameters
+   * @returns Promise<Key>
+   */
+  public async getKey(
+    keyName: string,
+    options?: GetKeyOptions
+  ): Promise<Key> {
+    const response = await this.client.getKey(
+      this.vaultBaseUrl,
+      keyName,
+      options && options.version ? options.version : "",
+      options
+    );
+    return this.getKeyFromKeyBundle(response);
+  }
+
+  /**
+   * The Get Deleted Key operation returns the specified deleted key along with its attributes.
+   * This operation requires the keys/get permission.
+   * @summary Gets the specified deleted key.
+   * @param keyName The name of the key.
+   * @param [options] The optional parameters
+   * @returns Promise<DeletedKey>
+   */
+  public async getDeletedKey(
+    keyName: string,
+    options?: RequestOptionsBase
```

Comment:
Perhaps for the methods which don't have any specific optional parameters we should have some sort of options base interface which just had the one property requestOptions?  This way the shape of the json you provide would be consistent for all the methods.  Thoughts?